### PR TITLE
Add query param for sorting search results

### LINKED ISSUE #904: Sort out the sort by story
- ~~disable sorting the doc id index.~~ => #908
- add a query param decouple the search sort with the indexing sort (right now they are strangely plugged via the IndexConfig)

Related to #860

### PR DESCRIPTION
**Pull request by PSeitz**
Fixes #904

### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

**Comment by fmassot** [comment]
Putting it here to not forget it: let's use the sort mini dsl for the sorting query param, see #860

File: quickwit-index-config/src/sort_by.rs [hunk 1]
```diff
@@ -52,7 +53,31 @@ impl From<SortOrder> for TantivyOrder {
...
+    /// Name of the field to sort by.
+    pub field_name: String,
+    /// Order to sort by. A usual top-k search implies a descending order.
+    pub order: SortOrder,
+}
+
+impl From<String> for SortByField {
+    fn from(string: String) -> Self {
+        let (field_name, order) = if string.starts_with('+') {
+            (string[1..].trim().to_string(), SortOrder::Asc)
```
Comment (fulmicoton) [inline quickwit-index-config/src/sort_by.rs:67]: this panics.

**Review by fulmicoton — APPROVED** [review]

File: quickwit-proto/src/cluster.rs [hunk 2]
```diff
@@ -4,284 +4,36 @@
...
...tCodec :: default () ; let mut grpc = tonic :: server :: Grpc :: new (codec) . apply_compression_config (accept_compression_encodings , send_compression_encodings) ; let res = grpc . unary (method , req) . await ; Ok (res) } ; Box :: pin (fut) } _ => Box :: pin (async move { Ok (http :: Response :: builder () . status (200) . header ("grpc-status" , "12") . header ("content-type" , "application/grpc") . body (empty_body ()) . unwrap ()) }) , } } } impl < T : ClusterService > Clone for ClusterServiceServer < T > { fn clone (& self) -> Self { let inner = self . inner . clone () ; Self { inner , accept_compression_encodings : self . accept_compression_encodings , send_compression_encodings : self . send_compression_encodings , } } } impl < T : ClusterService > Clone for _Inner < T > { fn clone (& self) -> Self { Self (self . 0 . clone ()) } } impl < T : std :: fmt :: Debug > std :: fmt :: Debug for _Inner < T > { fn fmt (& self , f : & mut std :: fmt :: Formatter < '_ >) -> std :: fmt :: Result { write ! (f , "{:?}" , self . 0) } } impl < T : ClusterService > tonic :: transport :: NamedService for ClusterServiceServer < T > { const NAME : & 'static str = "cluster.ClusterService" ; } }
```
Comment (fmassot) [inline quickwit-proto/src/cluster.rs:39]: you should not have merged this :/

File: quickwit-proto/src/lib.rs [hunk 3]
```diff
@@ -36,6 +36,8 @@ impl From<SearchStreamRequest> for SearchRequest {
             end_timestamp: item.end_timestamp,
             max_hits: 0,
             start_offset: 0,
+            sort_by_field: None,
```
Comment (fmassot) [inline quickwit-proto/src/lib.rs:39]: why not keeping the mini dsl on the grpc layer? I think it's better to keep it as we may handle multiple sort fields in the future. For now it's ok, so let's keep it like this.

**Pull request merged**